jib-core/src/main/java/com/google/cloud/tools/jib/configuration/LayerConfiguration.java

```diff
@@ -70,6 +70,21 @@ public Builder addEntry(Path sourceFile, AbsoluteUnixPath pathInContainer) {
       return this;
     }
 
+    /**
+     * Adds an entry to the layer. Only adds the single source file to the exact path in the
+     * container file system with the given permissions.
+     *
+     * @param sourceFile the source file to add to the layer
+     * @param pathInContainer the path in the container file system corresponding to the {@code
+     *     sourceFile}
+     * @param permissions the file permissions on the container
+     * @return this
+     */
+    public Builder addEntry(Path sourceFile, AbsoluteUnixPath pathInContainer, int permissions) {
```

Comment:
Hmm, maybe we should use `PosixFilePermission` instead since that is the class that represents file permissions.

Reply:
`PosixFilePermission` doesn't seem all that easy to use in our case. In the end we're going to be parsing a number from the build config, converting it into a `Set<PosixFilePermission>` to store, and then converting that set back to a number because there doesn't seem to be any way to set posix file permissions directly on `TarArchiveEntry`, the only place we actually need to use permissions.